src/iexamine.cpp

```diff
@@ -4255,6 +4327,45 @@ void smoker_activate( player &p, const tripoint &examp )
     add_msg( _( "You light a small fire under the rack and it starts to smoke." ) );
 }
 
+void iexamine::mill_finalize( player &, const tripoint &examp, const time_point &start_time )
+{
+    const furn_id cur_mill_type = g->m.furn( examp );
+    furn_id next_mill_type = f_null;
+    if( cur_mill_type == f_wind_mill_active ) {
+        next_mill_type = f_wind_mill;
+    } else if( cur_mill_type == f_water_mill_active ) {
+        next_mill_type = f_water_mill;
+    } else {
+        debugmsg( "Furniture executed action mill_finalize, but is of type %s",
+                  g->m.furn( examp ).id().c_str() );
+        return;
+    }
+
+    auto items = g->m.i_at( examp );
+    if( items.empty() ) {
+        g->m.furn_set( examp, next_mill_type );
+        return;
+    }
+
+    for( auto &it : items ) {
+        if( it.has_flag( "MILLABLE" ) ) {
+            it.calc_rot_while_processing( examp, 6_hours );
```

Comment:
This triggered a debugmsg:
``calc_rot_while_smoking called on non smoking item: wheat (frozen)``

Reply:
needed to be check for processing flag instead of just smoking flag .